Formula/cubelib.rb

```diff
@@ -0,0 +1,24 @@
+class Cubelib < Formula
+  desc "Cube, is a performance report explorer for Scalasca and Score-P"
+  homepage "https://scalasca.org/software/cube-4.x/download.html"
```

Comment:
Looking at this it seems like this tool isn't open source. Which would violate https://docs.brew.sh/Acceptable-Formulae#we-dont-like-binary-formulae is this the case?

Reply:
For cubelib : http://scalasca.org/software/cube-4.x/license.html
> The entire code of CUBE v4 is licensed under the BSD-style license agreement given below, except for the third-party code distributed in the 'vendor/' subdirectory. See the corresponding COPYING files in 'vendor/*' of the distribution tarball for details. 

In the vendor section I find 

1. gmock:
> Copyright (c) 2008 Google Inc.
> Copyright (c) 2013 Forschungszentrum Juelich GmbH
>
> All rights reserved.
>
>
> Redistribution and use in source and binary forms, with or without
> modification, are permitted provided that the following conditions are
> met:
>
> * Redistributions of source code must retain the above copyright
>   notice, this list of conditions and the following disclaimer.
>
> * Redistributions in binary form must reproduce the above copyright
>   notice, this list of conditions and the following disclaimer in the
>   documentation and/or other materials provided with the distribution.
> 
> * Neither the names of
>     Google Inc.,
>     the Forschungszentrum Juelich GmbH,
>   nor the names of their contributors may be used to endorse or promote
>   products derived from this software without specific prior written
>   permission.
> 
> THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
> "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
> LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
> A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
> OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
> SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
> LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
> DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
> THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
> (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
> OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

2. gtest: (same file)
3. common and windows: No Copying file.

Reply:
A BSD style license isn't really the same as an OSI approved license so unfortunately this software doesn't seem to qualify for homebrew-core. You could add it as a cask or a seperate tap though.

Reply:
Ok thanks. I'll keep it for me then.

Reply:
Running http://scalasca.org/software/cube-4.x/license.html through https://github.com/licensee/licensee unfortunately yields it does not match any known open source license with any confidence.

Reply:
I noticed the developers of that. Thanks.

Reply:
I was answered by the developerr:

> We are using the 3-clause BSD license which is OSI approved,
> see https://opensource.org/licenses/BSD-3-Clause

Reply:
@XL64 Is that documented in a `LICENSE.txt` or anywhere publicly?

Reply:
This BSD-3-Clause correspond to what can be found in COPYING files from cubelib, vendor/gmock and vendor/gtest

```
> diff --side-by-side -W 200 COPYING 3-Clause_BSD_License                                                                                                                     
                         CUBE v4 LICENSE AGREEMENT						   |	Copyright <YEAR> <COPYRIGHT HOLDER>
                         =========================						   <
												   <
Copyright (c) 1998-2017 Forschungszentrum Juelich GmbH, Germany					   <
Copyright (c) 2009-2015 German Research School for Simulation Sciences GmbH,			   <
                        Juelich/Aachen, Germany							   <
												   <
All rights reserved.										   <
												   <

Redistribution and use in source and binary forms, with or without					Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions are					modification, are permitted provided that the following conditions are
met:													met:

* Redistributions of source code must retain the above copyright				   |	1. Redistributions of source code must retain the above copyright
  notice, this list of conditions and the following disclaimer.					   |	notice, this list of conditions and the following disclaimer.
												   <
* Redistributions in binary form must reproduce the above copyright				   <
  notice, this list of conditions and the following disclaimer in the				   <
  documentation and/or other materials provided with the distribution.				   <

* Neither the names of										   |	2. Redistributions in binary form must reproduce the above copyright
    the Forschungszentrum Juelich GmbH,								   |	notice, this list of conditions and the following disclaimer in the
    the German Research School for Simulation Sciences GmbH,					   |	documentation and/or other materials provided with the distribution.
  nor the names of their contributors may be used to endorse or promote				   |
  products derived from this software without specific prior written				   |	3. Neither the name of the copyright holder nor the names of its
  permission.											   |	contributors may be used to endorse or promote products derived from
												   >	this software without specific prior written permission.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS					THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT					"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR					LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT					A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,				   |	HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT					SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,					LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY					DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT					THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE					(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.				   /	OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
```

Reply:
@XL64 I can see that they are similar but additions like `All rights reserved.` do change the meaning unless they have stated clearly and publicly (ideally in the tarball or license file) that this is a 3-Clause BSD license. Could you request that?

Reply:
I'll ask them, all I can find with a grep BSD on the code is "BSD-style license". So I don't think 3-Clause BSD strictly appears in the tarball.

Reply:
@XL64 Great, thanks. Sorry to be a pain on this just licensing is quite important to make sure us and our users aren't breaking the law.

Reply:
I understand. No problem.

Reply:
Here is the answer from the development team:

> i'm afraid that it doesn't state literally "BSD-3-Clause" or "3-Clause BSD" anywhere in tarball. Only if one compares wording of the license with https://opensource.org/licenses/BSD-3-Clause, one can see clearly that they are identical (up to the copyright holders)
> 
> Is it a show-stopper?

Reply:
@XL64 Unfortunately as you noted: they aren't identical as they have added `All rights reserved.` and other text. It is a show stopper if both we and e.g. https://github.com/licensee/licensee cannot verify this license.

Reply:
An other answer:
> The "All rights reserved" is (and always has been) part of the 3-clause
> BSD license, see also
>  - https://en.wikipedia.org/wiki/BSD_licenses#3-clause_license_(%22BSD_License_2.0%22,_%22Revised_BSD_License%22,_%22New_BSD_License%22,_or_%22Modified_BSD_License%22)
>  - https://spdx.org/licenses/BSD-3-Clause.html
> 
> The OSI page you mentioned is actually referring to SPDX, thus I don't
> see where the problem is.  But IANAL...